File: app/src/main/java/com/lostincontext/playlists/PlaylistsActivity.java
package com.lostincontext.playlists;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.FragmentManager;

import com.lostincontext.R;
import com.lostincontext.commons.BaseActivity;

import javax.inject.Inject;

import static com.lostincontext.utils.Activities.addFragmentToActivity;


public class PlaylistsActivity extends BaseActivity {


    @Inject PlaylistsPresenter presenter;


    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.playlists_screen);

        FragmentManager fm = getSupportFragmentManager();
        PlaylistsFragment fragment = (PlaylistsFragment) fm.findFragmentById(R.id.contentFrame);

        if (fragment == null) {
            fragment = PlaylistsFragment.newInstance();
            addFragmentToActivity(fm, fragment, R.id.contentFrame);
        }

        DaggerPlaylistsComponent.builder()
                .playlistsPresenterModule(new PlaylistsPresenterModule(fragment))
                .build()
                .inject(this);
    }
}

File: app/src/main/java/com/lostincontext/playlists/PlaylistsActivity.kt
package com.lostincontext.playlists

import android.os.Bundle
import com.lostincontext.R
import com.lostincontext.application.LostApplication
import com.lostincontext.commons.BaseActivity
import com.lostincontext.utils.addFragmentToActivity
import javax.inject.Inject


class PlaylistsActivity : BaseActivity() {


    @Inject lateinit internal var presenter: PlaylistsPresenter


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.playlists_screen)

        val fm = supportFragmentManager
        var fragment: PlaylistsFragment? = fm.findFragmentById(R.id.contentFrame) as PlaylistsFragment?

        if (fragment == null) {
            fragment = PlaylistsFragment.newInstance()
            addFragmentToActivity(fm, fragment, R.id.contentFrame)
        }

        DaggerPlaylistsComponent.builder()
                .playlistsPresenterModule(PlaylistsPresenterModule(fragment))
                .applicationComponent((application as LostApplication).appComponent)
                .build()
                .inject(this)
    }
}
